pkg/api/types.go

```diff
@@ -1205,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
```

Comment:
I just want the topology constraint to be required in the volume spec.  This is for security purposes, so if the scheduler makes some mistake or the user forgets to put the topology constraint in the PV, then we are not reverting back to hostpath behavior of mounting arbitrary directories on any node.

So I just want some way to be able to check at mount time, that the volume I am mounting is in fact owned by this node.

I think one way that we can generalize this, is instead of a nodeName field, have a topologyConstraints field that can look something like:

```
TopologyConstraints []TopologyConstraint

type TopologyConstrating struct {
    key string
    value string
}
```

The key-value would essentially be the allowed labels, like "hostname = node1", etc.

In the future, for some storage that could be local to 3 machines, as an example, then you would have 3 entries in this list.

WDYT?